### DISCUSSION TITLE
How to create archive, category pages

### DISCUSSION THREAD

**Discussion by mehidi258** [wpengine/faustjs#174] [Q&A] [answered]
I really liked this framework and i would like to use this one for my next project. 
I am struggling with handle archive, category pages.
So how i can use these template files inside theme directory and also if i want to custom-page-template then what to do here?

**Comment by trevanhetzel** [comment]
@mehidi258 Glad you're liking it so far! We are working this week on properly supporting archive and category templates. We'll have a fix and updated documentation soon. For the time being, you could create a [catch-all route](https://nextjs.org/docs/routing/dynamic-routes#optional-catch-all-routes) under `/pages` inside a directory that matches your permalink structure (`category` or `archives`, for example). You might still run into an issue getting data using the `getPosts` hook, however.

I'll update this issue when the fix for this is merged!

**Comment by mehidi258** [comment]
@trevanhetzel Thanks!
Glad to know this feature is coming soon!

**Comment by trevanhetzel** [comment]
@mehidi258 #133 was merged in last week and our getting-started example now has more fleshed out [examples](https://github.com/wpengine/headless-framework/blob/canary/examples/getting-started/wp-templates/category.tsx) of making custom queries and a category page example. There's also now a doc on [fetching data with custom queries](https://github.com/wpengine/headless-framework/tree/canary/docs/queries).

**Comment by mehidi258** [comment]
Awesome @trevanhetzel Really liked that and also thumbs up for the query part.
Last thing could you please tell me something about custom page template?

**Comment by trevanhetzel** [comment, marked as answer]
@mehidi258 If you're using a WP theme that has a custom page template, [templating in our framework](https://github.com/wpengine/headless-framework/tree/canary/docs/templating) works just like any other page type, following the [WP template hierarchy](https://developer.wordpress.org/themes/basics/template-hierarchy/). For instance, it will look for the ID, then slug, then a custom template.

If you're looking to render a template in Next.js based on a route, you can do that by simply creating a `.tsx` file in the `pages` directory with the name of the route. This will get hit instead of the `[[...page]].tsx` catch-all.

If you would rather stick with the catch-all route, you'd need to use something like [`useRouter`](https://nextjs.org/docs/api-reference/next/router) and render your custom template inside a conditional that matches the current route.

Hope that helps, but let me know if there are other specific custom page template questions!
